File: github-integration/src/main/java/de/zalando/zally/github/ApiValidationController.java
package de.zalando.zally.github;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ApiValidationController {

    private static final Logger LOG = LoggerFactory.getLogger(ApiValidationController.class);

    @ResponseBody
    @PostMapping("/api-validation")
    public void validatePullRequest(@RequestBody String payload) {

    }

}

File: github-integration/src/main/java/de/zalando/zally/github/ApiValidationController.kt
package de.zalando.zally.github

import de.zalando.zally.github.util.logger
import org.kohsuke.github.GHCommitState
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
class ApiValidationController(private val githubService: GithubService) {

    val log by logger()

    @ResponseBody
    @PostMapping("/api-validation")
    @ResponseStatus(value = HttpStatus.ACCEPTED)
    fun validatePullRequest(@RequestBody payload: String,
                            @RequestHeader(value = "X-GitHub-Event") eventType: String,
                            @RequestHeader(value = "X-Hub-Signature") signature: String) {

        if (eventType != ApiValidationController.PULL_REQUEST_EVENT_NAME) {
            log.info("Received unsupported event: {}", eventType)
            return
        }

        val pullRequest = githubService.parsePayload(payload, signature)

        pullRequest.updateCommitState(GHCommitState.FAILURE, "https://127.0.0.1", "Hello", "Test")
    }

    companion object {
        private val PULL_REQUEST_EVENT_NAME = "pull_request"
    }

}
